roles/ovb/bmc/tasks/main.yml

```diff
@@ -0,0 +1,8 @@
+# Setup BMC node
+- include: packages.yml
+- include: deploy_files.yml
+- include: os-net-config.yml
+
+# Deploy service file for each node in 'overcloud_nodes' group
+- include: openstack-bmc.yml
+  with_dict: "{{ provisioner.nodes }}"
```

Comment:
"why is this outside the play ?" - outside what play?

"you are gathering the cloud details for nothing for every component no?" not for nothing..for template.

"since you know the values you're looking for," - I don't know what values I'm looking for. I'm looking for every defined type of node.